registry/auth.go

```diff
@@ -232,7 +251,8 @@ func ResolveAuthConfig(config *cliconfig.ConfigFile, index *IndexInfo) cliconfig
 		stripped := url
 		if strings.HasPrefix(url, "http://") {
 			stripped = strings.Replace(url, "http://", "", 1)
-		} else if strings.HasPrefix(url, "https://") {
+		}
+		if strings.HasPrefix(url, "https://") {
```

Comment:
This is changing the behavior. This wasn't conditionally cleaning the url. If something like `http://https://something` is provided we should let them know instead of keep cleaning it to just `something`
Could you please revert this?